docs/docs/reference/contextual/context-functions.md

```diff
@@ -117,12 +117,12 @@ As a larger example, here is a way to define constructs for checking arbitrary p
 
 ```scala
 object PostConditions:
-   opaque type WrappedResult[T] = T
+   type WrappedResult[T] = T
```

Comment:
If it's just a type alias then WrappedResult[T] might as well be removed in favor of T, but I believe the point of using an opaque type was to make sure the `result` method below always returned the parameter coming from `ensuring` and not some random other implicit (e.g. in `1.ensure { given Int = 2; result == 1 }`, you want `result` to still be `1`.

Reply:
Yes, good point. If we have to pay a cost for the additional safety, then we can remove `zero-overhead abstraction` in the text.

Reply:
BTW, the safety is not bullet-proof -- it's possible to break it by confusing two `WrappedResult[Int]` via a mutable `var`. we have to rely on end-users to write reasonable code here.

Reply:
The idea is that `WrappedResult` is implementation internal. Use code should not mention it. In that case it is bullet-proof, I think.

Reply:
If we assume users cannot write the type explicitly in programmers, it's harder to break it. But it's still possible to do so, given that we have type inference and `export` mechanism.